examples/pytorch/multigpu/multi_gpu_link_prediction.py

```diff
@@ -0,0 +1,320 @@
+
+def evaluate(proc_id, model, g, device, use_uva):
+    model.eval()
+    batch_size = 10000
+    with torch.no_grad():
+        pred = model.module.inference(g, device, batch_size, use_uva)
+    return pred
+
+
+def train(
+        proc_id, nprocs, device, g, train_idx, val_idx, test_idx, model, use_uva
+):
+    # Create PyTorch DataLoader for constructing blocks
+    n_edges = g.num_edges()
+    train_seeds = torch.arange(n_edges).to(device)
+    labels = g.ndata["label"].to("cpu")
+
+    sampler = NeighborSampler([10, 25], prefetch_node_feats=["feat"])
+    sampler = as_edge_prediction_sampler(
+        sampler,
+        exclude="reverse_id",
+        # For each edge with ID e in Reddit dataset, the reverse edge is e ± |E|/2.
+        reverse_eids=torch.cat(
+            [torch.arange(n_edges // 2, n_edges),torch.arange(0, n_edges // 2)]
+        ).to(train_seeds),
+        # num_negs = 1, neg_share = False
+        negative_sampler=NegativeSampler(
+            g, 1, False, device if use_uva else None
+        ),
+    )
+    train_dataloader = DataLoader(
+        g,
+        train_seeds,
+        sampler,
+        device=device,
+        batch_size=10000,
+        shuffle=True,
+        drop_last=False,
+        num_workers=0,
+        use_ddp=True,
+        use_uva=use_uva,
+    )
+    opt = torch.optim.Adam(model.parameters(), lr=0.003)
+    loss_fcn = CrossEntropyLoss()
+    iter_pos = []
+    iter_neg = []
+    for epoch in range(10):
+        tic = time.time()
+        model.train()
+        for step, (input_nodes, pos_graph, neg_graph, blocks) in enumerate(
+                train_dataloader
+        ):
+            x = blocks[0].srcdata["feat"]
+            y_hat = model(blocks, x)
+            loss = loss_fcn(y_hat, pos_graph, neg_graph)
+            opt.zero_grad()
+            loss.backward()
+            opt.step()
+
+            if step % 20 == 0 and proc_id == 0: # log every 20 steps
+                # gpu memory reserved by PyTorch
+                gpu_mem_alloc = (
+                    torch.cuda.max_memory_allocated() / 1000000
+                    if torch.cuda.is_available()
+                    else 0
+                )
+                print(
+                    "Epoch {:05d} | Step {:05d} | Loss {:.4f} | GPU {:.1f} MB".format(
+                        epoch, step, loss.item(), gpu_mem_alloc
+                    )
+                )
+
+        toc = time.time()
+        if proc_id == 0:
+            print("Epoch Time(s): {:.4f}".format(toc - tic))
+        if (epoch + 1) % 5 == 0: # eval every 5 epochs
+            pred = evaluate(proc_id, model, g, device, use_uva) # in parallel
+            if proc_id == 0:
+                # only master proc does the accuracy computation
+                eval_acc, test_acc = compute_acc_unsupervised(
+                    pred, labels, train_idx, val_idx, test_idx
+                )
+                print(
+                    "Epoch {:05d} | Eval Acc {:.4f} | Test Acc {:.4f}".format(
```

Comment:
The results are "F1 score" rather than "accuracy". Let's use the correct terminology.